src/poetry/utils/helpers.py

```diff
@@ -312,3 +334,14 @@ def extractall(source: Path, dest: Path, zip: bool) -> None:
                 archive.extractall(dest, filter="data")
             else:
                 archive.extractall(dest)
+
+
+def get_highest_priority_hash_type(hash_types: set[str]) -> str | None:
+    if not hash_types:
+        return None
+
+    for prioritised_hash_type in prioritised_hash_types:
+        if prioritised_hash_type in hash_types:
+            return prioritised_hash_type
+
+    return None
```

Comment:
```suggestion

    logger.debug('None of the hash types %s is in prioritized_hash_types', hash_types)

    for hash_type in hash_types:
        if hash_type in non_prioritized_hash_types:
            return hash_type

    logger.warning('None of the hash types %s is available in hashlib', hash_types)

    return None
```

If there is no prioritized hash type we just "randomly" choose an available hash type. That's why md5 and sha1 can be omitted in `prioritized_hash_types`.

If none of the hash types is available we probably should log a warning.

Reply:
Cool, I see what your saying now. I'll create a commit for this...